workspaces/src/network/testnet.rs

```diff
@@ -0,0 +1,109 @@
+use std::fs::File;
+use std::io::Read;
+use std::path::{Path, PathBuf};
+use std::str::FromStr;
+
+use async_trait::async_trait;
+use url::Url;
+
+use near_crypto::{InMemorySigner, Signer};
+use near_primitives::{types::AccountId, views::FinalExecutionStatus};
+
+use crate::network::{
+    Account, AllowDevAccountCreation, CallExecution, CallExecutionDetails, NetworkClient,
+    NetworkInfo, TopLevelAccountCreator,
+};
+use crate::rpc::{client::Client, tool};
+use crate::Contract;
+
+const RPC_URL: &str = "https://rpc.testnet.near.org";
+const HELPER_URL: &str = "https://helper.testnet.near.org";
+
+pub struct Testnet {
+    client: Client,
+}
+
+impl Testnet {
+    pub(crate) fn new() -> Self {
+        Self {
+            client: Client::new(RPC_URL.into()),
+        }
+    }
+}
+
+impl AllowDevAccountCreation for Testnet {}
+
+#[async_trait]
+impl TopLevelAccountCreator for Testnet {
+    async fn create_tla(
+        &self,
+        id: AccountId,
+        signer: InMemorySigner,
+    ) -> anyhow::Result<CallExecution<Account>> {
+        tool::url_create_account(
+            Url::parse(&self.helper_url())?,
+            id.clone(),
+            signer.public_key(),
+        )
+        .await?;
+
+        Ok(CallExecution {
+            result: Account::new(id, signer),
+            details: CallExecutionDetails {
+                // We technically have not burnt any gas ourselves since someone else paid to
+                // create the account for us in testnet when we used the Helper contract.
+                total_gas_burnt: 0,
+
+                status: FinalExecutionStatus::SuccessValue(String::new()),
+            },
+        })
+    }
+
+    async fn create_tla_and_deploy<P: AsRef<Path> + Send + Sync>(
+        &self,
+        id: AccountId,
+        signer: InMemorySigner,
+        wasm: P,
+    ) -> anyhow::Result<CallExecution<Contract>> {
+        // TODO: async_compat/async version of File
+        let mut code = Vec::new();
+        File::open(wasm)?.read_to_end(&mut code)?;
+
+        let account = self.create_tla(id.clone(), signer.clone()).await?;
+        let account = Into::<anyhow::Result<_>>::into(account)?;
```

Comment:
Just noting there should probably be a more convenient API for converting into a result. Maybe we can add a `into_result` function on the `CallExecution`?

Reply:
Sounds good with me